flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/functions/CastFunctionITCase.java

```diff
@@ -18,176 +18,905 @@
+                                LocalDate.parse("2021-09-24"),
+                                LocalDateTime.of(2021, 9, 24, 0, 0, 0, 0))
+                        // https://issues.apache.org/jira/browse/FLINK-24415 Fractional seconds are
+                        // lost
+                        // https://issues.apache.org/jira/browse/FLINK-24423 Continue using EPOCH
+                        // date or use 0 for the year?
+                        .testCase(
+                                TIME(5),
+                                LocalTime.parse("12:34:56.1234567"),
+                                LocalDateTime.of(1970, 1, 1, 12, 34, 56, 0))
+                        .testCase(
+                                TIMESTAMP(),
+                                LocalDateTime.parse("2021-09-24T12:34:56.123456"),
+                                LocalDateTime.of(2021, 9, 24, 12, 34, 56, 123456000))
+                        .testCase(
+                                TIMESTAMP(4),
+                                LocalDateTime.parse("2021-09-24T12:34:56.123456"),
+                                LocalDateTime.of(2021, 9, 24, 12, 34, 56, 123400000))
+                        // Not supported currently
+                        // TIMESTAMP_WITH_TIME_ZONE
+                        // https://issues.apache.org/jira/browse/FLINK-24422 - Accept only Instant
+                        .testCase(
+                                TIMESTAMP_LTZ(4),
+                                LocalDateTime.parse("2021-09-24T12:34:56.123456"),
+                                LocalDateTime.of(2021, 9, 24, 12, 34, 56, 123400000))
+                        .testCase(
+                                TIMESTAMP_LTZ(4),
+                                LocalDateTime.parse("2021-09-24T22:34:56.123456")
+                                        .toInstant(ZoneOffset.ofHours(5)),
+                                LocalDateTime.of(2021, 9, 25, 1, 34, 56, 123400000))
+                        // Not supported - no fix
+                        // INTERVAL(YEAR())
+                        // INTERVAL(DAY())
+                        // ARRAY
+                        // MULTISET
+                        // MAP
+                        // ROW
+                        // RAW
+                        .build(),
+                CastTestSpecBuilder.test(TIMESTAMP_LTZ(9), "To TIMESTAMP_LTZ")
+                        // Causes NPE
+                        // .testCase(CHAR(3), "foo", null)
+                        // .testCase(VARCHAR(5), "Flink", null)
+                        // .testCase(STRING(), "123", null)
+                        .testCase(
+                                STRING(),
+                                "2021-09-27",
+                                fromLocalToUTC(LocalDateTime.of(2021, 9, 27, 0, 0, 0, 0)))
+                        // https://issues.apache.org/jira/browse/FLINK-24415 Fractional seconds are
+                        // lost
+                        .testCase(
```

Comment:
this is not related to FLINK-24415, we are using TIMESTAMP not TIME here

Reply:
Thx, created a new issue for that [FLINK-24446](https://issues.apache.org/jira/browse/FLINK-24446)